components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/GeoLocationBasedServiceImpl.java

```diff
@@ -178,9 +178,40 @@ public Response createGeoAlerts(Alert alert, @PathParam("deviceId") String devic
             String error = "Error occurred while creating the geo alert for " + deviceType + " with id: " + deviceId;
             log.error(error, e);
             return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(error).build();
+        } catch (AlertAlreadyExistException e) {
+            String error = "A geo alert with this name already exists.";
+            log.error(error,e);
+            return Response.status(Response.Status.BAD_REQUEST).entity(error).build();
         }
     }
 
+
+    @Path("alerts/{alertType}")
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Response createGeoAlertsForGeoClusters(Alert alert, @PathParam("alertType") String alertType) {
+        try {
+            // this is the user who initiates the request
+            String authorizedUser = MultitenantUtils.getTenantAwareUsername(
```

Comment:
Where do you use this assignment to the local variable 'authorizedUser'?

Reply:
Removed the unused variable. c653051